### DISCUSSION TITLE
Why PascalCase?

### DISCUSSION THREAD

**Discussion by 1vanK** [gobolinux/Documentation#26] [Q&A]
In Linux it is common practice to use the snake_case, since FS is case sensitive. Why are you using PascalCase?

**Comment by lopiuh** [comment]
My 2cents (generated with the help of google ai studio)

GoboLinux uses PascalCase (or capitalization) for both the root directory structure and package names to enhance readability and present a visually cleaner, more logical system.

While standard Linux follows the FHS with lowercase, often cryptic abbreviations (like `/usr`, `/var`, `/etc`), GoboLinux uses full, human-readable words:
*   `/Programs`
*   `/System`
*   `/Users`
*   `/Mount`

Inside `/Programs`, directories usually follow the "Brand Name" of the software (e.g., `/Programs/Firefox` or `/Programs/Xorg-Server`).

This design choice was historically inspired by systems like BeOS (and partly macOS), prioritizing a clean structure over traditional Unix legacy conventions. While it is untypical for Linux, modern shell tab-completion usually makes the case-sensitivity issue negligible in daily use.

**Comment by rubyFeedback** [comment]
> In Linux it is common practice to use the snake_case, since FS is case sensitive. Why are you using PascalCase?

You refer to the names of programs, right? Such as:

   /Programs/KDE-Libs/

Or names like that?

I think this is in part historic; and also a preference by hisham. Personally I don't really care either way; in my
own ruby scripts I simplified it to simply first-character uppercased, rest downcased, all '-' or '_' or '.' removed,
so the above would then be: /Programs/Kdelibs/. For some reason I prefer the first letter uppercased. I
also experimented with other things such as /pkg/kdelibs/. Homebrew uses I think /usr/cellar/ or
/usr/local/cellar/ or Cellar/ or something like that. Ultimately, though, I actually think the names are not
that important as such - the more important part is that things have their own, ideally versioned, AppDir.
NixOS uses some hashed names - I forgot the exact name, but https://discourse.nixos.org/t/how-is-the-nix-hash-for-a-directory-generated/18957/5 referred to it).

If you refer to / and /usr then actually they don't use any '_' there as far as I remember, and GoboLinux has
the compatibility symlinks anyway, so GoboLinux is conforming to the FHS actually - see hisham's old
https://gobolinux.org/doc/articles/clueless.html.

In my own scripts I try to let the user decide what to use, but the most tested variant I use is the one
I described above, largely because I can not remember any rules for determining this automatically.

Of course one question here may still be whether you are fine with capitalized names for directories.
If you refer to Linux in general, most directories are downcased. But I think the primary reason for
this is laziness and convenience; it is easier to type a downcased character. The FHS is not really 
one of "epic design" but mostly a post-hoc "standard", in my opinion. It described the common 
practice, not necessarily logic (why is /usr/ really existing ... and does /usr/local/ really make sense
if a system uses versioned appdirs already?).

Edit: Actually, SnakeNames are not only about the uppercased letter, but also not using '-' or '_',
so I guess you are fine with '_' but not uppercased letters? How about '-' - is this ok or not ok to
use? Using '_' is actually less convenient than just '-' on a keyboard. I am not sure why '_' became
more frequently used than '-'. I totally understand that lowercased directory names are easier
to type. On GoboLinux, /mnt/ is also a symlink towards /Mount right? So "cd /mnt" should work
fine. I guess the only confusing part may be the /System/ ... something for mounted devices,
such as sda1 and so forth. Do we use a compatibility layer for that? I forgot. On most linux we
have e. g. /dev/sda1 or similar.

**Comment by 1vanK** [comment]
It's not that it's easier to type, but that folders named "folder," "FOLDER," and "Folder" can all exist same time. And when it's standard practice to use lowercase letters in names, this situation is eliminated.
